src/core/usecases/runningService.ts

```diff
@@ -135,6 +140,38 @@ export const thunks = {
 
             dispatch(actions.isUserWatchingChanged({ isUserWatching }));
         },
+    "upgradeService":
+        (serviceId: string): ThunkAction<void> =>
+        async (...args) => {
+            const [dispatch, getState] = args;
+
+            {
+                const state = getState().runningService;
+
+                if (state.isUpdating) {
+                    return;
+                }
+            }
+            const restorableRunningPackageConfigs =
+                runningPackageConfigSelectors.restorableRunnigPackageConfigs(getState());
+            const restorableRunningPackageConfig = restorableRunningPackageConfigs.find(
+                e => e.runningPackageConfig.name === serviceId
+            )?.runningPackageConfig;
+            assert(restorableRunningPackageConfig !== undefined);
+
+            await dispatch(
+                launcherThunk.initialize({
+                    catalogId: restorableRunningPackageConfig.catalogId,
+                    packageName: restorableRunningPackageConfig.packageName,
+                    name: restorableRunningPackageConfig.name,
+                    formFieldsValueDifferentFromDefault:
+                        restorableRunningPackageConfig.formFieldsValueDifferentFromDefault
+                })
+            );
+            await dispatch(launcherThunk.launch());
+            dispatch(launcherThunk.reset());
+            dispatch(thunks.update());
```

Comment:
This is hacky.  
It's like simulating buttons clicks.  
This thunks has been desined specifically as button callbacks for the UI, not as functions that can be composed to acheive other behaviour.  

I'm going to approve this anyway...